contracts/test/ocr2/aggregator.test.ts

```diff
@@ -257,70 +367,109 @@ describe('Aggregator', function () {
+            transmitter: oracle.transmitter.address,
+            payee: oracle.transmitter.address, // reusing transmitter acocunts as payees for simplicity
+          })),
+        }),
       })
+
       // several rounds happened so we are owed payment
-      assert.ok(owed > 0)
+      const payee = oracles[0].transmitter
+      const [owed1] = await provider.callContract({
+        contractAddress: aggregator.deploy.address,
+        entrypoint: 'owed_payment',
+        calldata: CallData.compile({
+          transmitter: payee.address,
+        }),
+      })
+      assert.ok(parseInt(owed1, 16) > 0)
 
       const availableToValue = ([is_negative, abs_difference]: [boolean, bigint]): bigint => {
         return is_negative ? -abs_difference : abs_difference
       }
 
       // no funds on contract, so no LINK available for payment
-      let { response: available } = await aggregator.call('link_available_for_payment')
-      assert.ok(availableToValue(available) < 0) // should be negative: we owe payments
+      let [isNegative, absDiff] = await provider.callContract({
+        contractAddress: aggregator.deploy.address,
+        entrypoint: 'link_available_for_payment',
+      })
+      assert.ok(availableToValue([isNegative === '0x1', BigInt(absDiff)]) < 0) // should be negative: we owe payments
 
       // deposit LINK to contract
-      await owner.invoke(token, 'transfer', {
-        recipient: aggregator.address,
-        amount: 100_000_000_000,
+      await owner.execute({
+        contractAddress: token.deploy.address,
+        entrypoint: 'transfer',
+        calldata: CallData.compile({
+          recipient: aggregator.deploy.address,
+          amount: cairo.uint256(100_000_000_000n),
+        }),
       })
 
       // we have enough funds available now
-      available = (await aggregator.call('link_available_for_payment')).response
-      assert.ok(availableToValue(available) > 0)
+      ;[isNegative, absDiff] = await provider.callContract({
+        contractAddress: aggregator.deploy.address,
+        entrypoint: 'link_available_for_payment',
+      })
+      assert.ok(availableToValue([isNegative === '0x1', BigInt(absDiff)]) > 0)
 
       // attempt to withdraw the payment
-      await payee.invoke(aggregator, 'withdraw_payment', {
-        transmitter: oracle.transmitter.starknetContract.address,
+      await payee.execute({
+        contractAddress: aggregator.deploy.address,
+        entrypoint: 'withdraw_payment',
+        calldata: CallData.compile({
+          transmitter: payee.address,
+        }),
       })
 
       // balance as transferred to payee
-      let { response: balance } = await token.call('balance_of', {
-        account: payee.starknetContract.address,
+      const [balance] = await provider.callContract({
+        contractAddress: token.deploy.address,
+        entrypoint: 'balance_of',
+        calldata: CallData.compile({
+          account: payee.address,
+        }),
       })
```

Comment:
Similarly the contract wrappers should have direct interfaces, not needing callContract/entrypoint/calldata

https://www.starknetjs.com/docs/next/guides/interact#-read-from-contract-memory-with-meta-class

Reply:
I've replaced all occurrences of `callContract` with the direct interface - thx for the pointer!